dashboard/app/models/levels/level.rb

```diff
@@ -43,6 +43,8 @@ class Level < ActiveRecord::Base
   has_many :levels_child_levels, -> {order('position ASC')}, class_name: 'ParentLevelsChildLevel', foreign_key: :parent_level_id
   has_many :child_levels, through: :levels_child_levels, inverse_of: :parent_levels
 
+  has_one :contained_level, -> {where(kind: 'contained')}, through: :levels_child_levels, source: :child_level
```

Comment:
somehow, this simpler syntax for contained_level works! the where clause refers to a field on the join table, and rails is able to figure that out:
```
p = Level.last
c = Level.first
p.contained_level = c
c.levels_parent_levels.first
=> #<ParentLevelsChildLevel id: 3, parent_level_id: 23369, child_level_id: 1, position: nil, kind: "contained">
```